operations/CSM_product_management/Configure_Non-Compute_Nodes_with_CFS.md

```diff
@@ -32,249 +50,367 @@ own keys, or use their own solution for authentication.
 
 ### Option 1: Use the CSM-provided SSH Keys
 
 The default CSM Ansible plays are already configured to enable Passwordless SSH
 by default. No further action is necessary before running NCN personalization
 with CFS.
 
+<a name="set_up_passwordless_ssh_option_2"></a>
+
 ### Option 2: Provide Custom SSH Keys
 
 Administrators may elect to replace the CSM-provided keys with their own custom
-keys. The `replace_ssh_keys.sh` script can be used to replace the keys from
-files.
+keys.
 
-```bash
-ncn-m001# /usr/share/doc/csm/scripts/operations/configuration/replace_ssh_keys.sh \
---public-key-file ./id_rsa-csm.pub --private-key-file ./id_rsa-csm
-```
+1. Set variables to the locations of the public and private SSH key files.
 
-Alternatively, the keys stored in Kubernetes can be updated directly.
+    > Replace the values in the examples below with the paths to the desired
+    > key files on the system.
 
-1. Replace the private key half:
-   ```bash
-   ncn# kubectl get secret -n services csm-private-key -o json | jq --arg value "$(cat ~/.ssh/id_rsa-csm | base64)" '.data["value"]=$value' | kubectl apply -f -
-   ```
-   where `~/.ssh/id_rsa-csm` is a local file containing a valid SSH private key.
+    ```bash
+    ncn# PUBLIC_KEY_FILE=/root/.ssh/id_rsa-csm.pub
+    ncn# PRIVATE_KEY_FILE=/root/.ssh/id_rsa-csm
+    ```
 
-1. Replace the public key half:
-   ```bash
-   ncn# kubectl delete configmap -n services csm-public-key && \
-      cat ~/.ssh/id_rsa-csm.pub | \
-      base64 > ./value && kubectl create configmap --from-file \
-      value csm-public-key --namespace services && rm ./value
-   ```
-   where `~/.ssh/id_rsa-csm.pub` is a local file containing a valid public key
-   intended for CSM and downstream products.
+1. Provide the custom keys by script or manually.
+
+    There are two options for providing the keys.
+
+    - Provide custom SSH keys by script.
+
+        The `replace_ssh_keys.sh` script can be used to replace the keys from files.
+
+        > The `docs-csm` RPM must be installed in order to use this script. See
+        > [Check for Latest Documentation](../../update_product_stream/index.md#documentation)
+
+        ```bash
+        ncn# /usr/share/doc/csm/scripts/operations/configuration/replace_ssh_keys.sh \
+                --public-key-file "${PUBLIC_KEY_FILE}" --private-key-file "${PRIVATE_KEY_FILE}"
+        ```
+
+    - Manually provide custom SSH keys
+
+        The keys stored in Kubernetes can be updated directly.
+
+        1. Replace the private key half:
+
+            ```bash
+            ncn# KEY64=$(cat "${PRIVATE_KEY_FILE}" | base64) &&
+                 kubectl get secret -n services csm-private-key -o json | \
+                    jq --arg value "$KEY64" '.data["value"]=$value' | kubectl apply -f - &&
+                 unset KEY64
+            ```
+
+        1. Replace the public key half:
+
+            ```bash
+            ncn# kubectl delete configmap -n services csm-public-key && \
```

Comment:
I don't believe you need the trailing /\'s on line 112-114.

Reply:
Right you are!